src/MogileFS/File/Mapper.php

```diff
@@ -162,6 +162,44 @@ public function fetchFile(MogileFS_File $file)
 		return $file;
 	}
 
+    public function fetchRawContent(MogileFS_File $file)
+    {
+        if (!$file->isValid()) {
+            throw new MogileFS_Exception(__METHOD__ . ' Cannot fetch file from invalid file model',
+                MogileFS_Exception::INVALID_ARGUMENT);
+        }
+
+        $paths = $file->getPaths();
+        $url = reset($paths);
+
+        $ch = curl_init($url);
+        $bodyStream = fopen('php://temp', 'w+');
+
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
```

Comment:
Returning transfer in addition to writing to file handle? Seems you're putting two copies of the downloaded file in memory now before returning!